packages/dd-trace/src/config.js

```diff
@@ -175,6 +191,37 @@ class Config {
       process.env.DD_TRACE_EXPERIMENTAL_TRACEPARENT_ENABLED,
       false
     )
+    // TODO: change this to traceparent,datadog
+    // Traceparent should be prioritized over datadog headers, but this is a breaking change until
+    // we support tracestate headers too, otherwise we would lose origin and tags header data.
+    const defaultPropagationStyle = ['datadog']
+    if (isTrue(DD_TRACE_TRACEPARENT_ENABLED)) {
+      defaultPropagationStyle.push('tracecontext')
+    }
+    if (isTrue(DD_TRACE_B3_ENABLED)) {
+      defaultPropagationStyle.push('b3')
+      defaultPropagationStyle.push('b3 single header')
+    }
+    if (process.env.DD_TRACE_PROPAGATION_STYLE && (
+      process.env.DD_TRACE_PROPAGATION_STYLE_INJECT ||
+      process.env.DD_TRACE_PROPAGATION_STYLE_EXTRACT
+    )) {
+      log.warn(
```

Comment:
Yes, because the priority ordering may be misunderstood by the user so it's part of the spec to warn if both are used together. There's not really any reason for the two to be used together given that the union of inject and extract is functionally equal to the shared version.